openedx-unsupported/configuration
# [SE-4612] Cherry-pick #6482 into lilac.master

## Description

Cherry-picks #6482 into `open-release/lilac.master`

## Supporting information
**Sandbox Link**: [SE-4612: (Lilac) Discussion Forum / ES Configs](https://manage.opencraft.com/instance/29210/edx-appserver/20051/)
**Sandbox Server IP**: 149.202.161.121

## Testing instructions
Since this was an issue that occurred when configuring new servers, you could launch a new server to make sure everything is fine. However the sandbox I've provided is a freshly configured one without any manual intervention so it should suffice to check that.

#### Testing by using the forums
We can test that elasticsearch is running be testing the discussion forums, since this forums won't work if ES is not running:

- Go to the [course page](https://se-4612-lilac.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/discussion/forum/) on the sandbox instance
- Click on the 'Discussions' link
- The page should open without errors, creating and searching for posts should work

#### Testing by checking heartbeat

- Visit the [heartbeat URL](https://se-4612-lilac.opencraft.hosting/heartbeat?extended) on the sandbox server and check that all endpoints are healthy 

#### Testing on the server
If you have SSH access to the server (I can give you access if you need it):

- Check that the elasticsearch service is running (systemctl status elasticsearch)
- Check that the elasticsearch process is running with the right configurations. To do this check it's enough to check elasticsearch is running with a 512m heapsize rather than the default 1g. You can check this by running the command below:
    ```
    ps aux | grep elasticsearch | awk 'match($0,/-Xmx[0-9]+m/) {print substr($0,RSTART,RLENGTH)}'
    ```
    If it outputs `-Xmx512m` then we're good, if it outputs `-Xmx1g` then it's using the wrong configurations.

## Reviewers
